### PR TITLE
Add support for 'memstore', an in-memory session store

### PR DESCRIPTION
**Pull request by nerney**
While a cookie store keeps the session information in the cookies themselves, the other store options require use of a third-party service to store session data. 'memstore' is an in-memory store that stores session data in an in-memory map. 'memstore' is concurrency-safe. Perfect for testing/dev environments. Or, use-cases where simple cookie sessions are desired with session data server-side rather than in the cookies, and possible loss of sessions is not catastrophic.

Added a sub-packages 'memstore' and 'example-memstore'.
Added 'memstore' to readme and other documentation along with other stores.
Added 'memstore' to the vendors.

see - https://github.com/quasoft/memstore

### CODE REVIEW

**Comment by nerney** [comment]
@appleboy @delphinus

File: README.md [hunk 1]
```diff
@@ -28,7 +28,8 @@ import "github.com/gin-contrib/sessions"
 
 #### cookie-based
 
-[embedmd]:# (example_cookie/main.go go)
+[embedmd]: # "example_cookie/main.go go"
+
```
Comment (delphinus) [inline README.md]: Are these necessary diffs?  I think `[embedmd]: # (...)` is a special format for [embedmd](https://github.com/campoy/embedmd).

File: vendor/vendor.json [hunk 2]
```diff
@@ -1,169 +1,175 @@
...
-		},
-		{
-			"checksumSHA1": "fALlQNY1fM99NesfLJ50KguWsio=",
-			"path": "gopkg.in/yaml.v2",
-			"revision": "cd8b52f8269e0feb286dfeef29f8fe4d5b397e0b",
-			"revisionTime": "2017-04-07T17:21:22Z"
-		}
-	],
-	"rootPath": "github.com/gin-contrib/sessions"
+  "comment": "",
```
Comment (delphinus) [inline vendor/vendor.json]: Indents changed here (tab => 2 spaces?).  govendor did this?

File: .gitignore [hunk 3]
```diff
@@ -1,3 +1,3 @@
 coverage.out
 vendor/*
-!/vendor/vendor.json
+!/vendor/vendor.json
```
Comment (delphinus) [inline .gitignore]: This diff seems also unnecessary.

File: README.md (hunk 1 above)
Comment (delphinus) [reply README.md]: Ah, you did this commit for travis!  See [the log](https://travis-ci.org/gin-contrib/sessions/jobs/375171855), it says



This reports it discovered a redundant empty line here in README.  Fix it.

File: .gitignore (hunk 3 above)
Comment (appleboy) [reply .gitignore]: revert this changes.

File: vendor/vendor.json (hunk 2 above)
Comment (appleboy) [reply vendor/vendor.json]: please update this file via `govendor` command.

**Comment by nerney** [comment]
alright, @appleboy
i removed the whitespace change from .gitignore.
i reverted vendor.json and then used govendor to add the new dep, no more formatting stuff from my IDE.
i fixed the changes to readme that VSCode had made, embedmd syntax reverted and the new memstore section conforms with the existing syntax.

File: .gitignore (hunk 3 above)
Comment (nerney) [reply .gitignore]: done

File: README.md (hunk 1 above)
Comment (nerney) [reply README.md]: fixed

File: vendor/vendor.json (hunk 2 above)
Comment (nerney) [reply vendor/vendor.json]: i fixed all this

**Pull request merged**